msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerAuthServiceStrategy.java

```diff
@@ -141,17 +144,25 @@ AcquireTokenResult acquireTokenSilent(AcquireTokenSilentOperationParameters para
         Telemetry.emit(
                 new BrokerStartEvent()
                         .putAction(methodName)
+                        .putStrategy(TelemetryEventStrings.Value.BOUND_SERVICE)
         );
 
         IMicrosoftAuthService service;
-
         MicrosoftAuthClient client = new MicrosoftAuthClient(parameters.getAppContext());
         MicrosoftAuthServiceFuture future = client.connect();
 
         try {
             //Do we want a time out here?
             service = future.get();
-        } catch (Exception e) {
+        } catch (final InterruptedException | ExecutionException e) {
+            Telemetry.emit(
+                    new BrokerEndEvent()
+                            .putAction(methodName)
+                            .isSuccessful(false)
+                            .putErrorCode(ErrorStrings.IO_ERROR)
+                            .putErrorDescription(e.getLocalizedMessage())
```

Comment:
In some cases, we're putting a generic constant string here. In this case, we're using the error message from the exception object. What's the reasoning behind where to put what?

(Personally I think putting exception message might be helpful, but I assume that we are putting generic message here to make sure we're not leaking PII).

Reply:
This is aligned with ADAL and other platforms.

Reply:
I'm seeing conflicting implementation here and in the line I commented above (that one uses a constant string), hence the question :)

Reply:
Key.ERROR_DESCRIPTION is alwasy treated as oii and we filtered them out in both logging and telemetry when pii/oii is disabled.